src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -157,10 +162,29 @@ int enroll_tpm2(struct crypt_device *cd,
                         return r;
         }
 
+        r = tpm2_load_pcr_public_key(pubkey_path, &pubkey, &pubkey_size);
+        if (r < 0) {
+                if (pubkey_path || signature_path || r != -ENOENT)
+                        return log_error_errno(r, "Failed read TPM PCR public key: %m");
+
+                log_debug_errno(r, "Failed to read TPM2 PCR public key, proceeding without: %m");
+                pubkey_pcr_mask = 0;
+        } else {
+                /* Also try to load the signature JSON object, to verify that our enrollment will work. This is optional however. */
+
+                r = tpm2_load_pcr_signature(signature_path, &signature_json);
+                if (r < 0) {
+                        if (signature_path || r != -ENOENT)
+                                return log_debug_errno(r, "Failed to read TPM PCR signature: %m");
+
+                        log_debug_errno(r, "Failed to read TPM2 PCR signature, proceeding without: %m");
```

Comment:
shouldn't this be info at least?

Reply:
This line is only hit if the user did not specify an explicit signature, and we then searched for one installed in the fs somewhere, and didn't find any. This is actually the common case on existing systems: they won't have a signature file installed, because simply noone builds systems with signed PCR values yet.

Hence, this really *is* a debugging thing: it will tell you explicitly that we failed to auto-discover a signature file then.